### PR TITLE
Save Completed Campaign Scenario to file

### PR DESCRIPTION
**Pull request by eos428**
While testing campaign awards, @Branikolog remarked that in the OG, when we complete a campaign scenario the OG creates a save file which will immediately lead you to the campaign scenario selection window (or probably the credits, if it's the last one).

I'm pretty sure this will have merge conflicts with the big campaign award PR, so when either of the PRs are merged I will resolve the conflicts asap.

Example: Good_1_Complete

### CODE REVIEW

File: src/fheroes2/game/game_campaign.cpp [hunk 1]
```diff
@@ -28,9 +28,11 @@
 #include "cursor.h"
 #include "dialog.h"
 #include "game.h"
+#include "game_io.h"
 #include "icn.h"
 #include "race.h"
 #include "settings.h"
+#include "system.h"
```
Comment (ihhub) [inline src/fheroes2/game/game_campaign.cpp:35]: Do we need this file inclusion?

File: src/fheroes2/game/game_io.cpp [hunk 2]
```diff
@@ -126,6 +126,19 @@ bool Game::AutoSave()
     return Game::Save( System::ConcatePath( GetSaveDir(), "AUTOSAVE" + GetSaveFileExtension() ) );
 }
 
+bool Game::SaveCompletedCampaignScenario()
+{
+    const std::string & name = Settings::Get().CurrentFileInfo().name;
+
+    std::string base = name.size() ? name : "newgame";
```
Comment (ihhub) [inline src/fheroes2/game/game_io.cpp:133]: Please use translation code `_( newgame )`.

File: src/fheroes2/game/game_io.cpp [hunk 3]
```diff
@@ -126,6 +126,19 @@ bool Game::AutoSave()
...
+    base.resize( std::distance( base.begin(), std::find_if( base.begin(), base.end(), ::ispunct ) ) );
```
Comment (ihhub) [inline src/fheroes2/game/game_io.cpp:134]: Are we going to remove a dot *.* here from the end? If yes please use `rfind` call instead.

File: src/fheroes2/game/game_io.cpp [hunk 4]
```diff
@@ -126,6 +126,19 @@ bool Game::AutoSave()
...
+    std::replace_if( base.begin(), base.end(), ::isspace, '_' );
+    std::ostringstream os;
+
+    os << System::ConcatePath( Game::GetSaveDir(), base ) << "_Complete" << Game::GetSaveFileExtension();
```
Comment (ihhub) [inline src/fheroes2/game/game_io.cpp:138]: Please use translation code `_( _Complete )`. Also string concatenation works totally fine with `std::string`. We do not need **ostringstream** here.

File: src/fheroes2/game/game_io.cpp [hunk 5]
```diff
@@ -126,6 +126,19 @@ bool Game::AutoSave()
...
+    return Game::Save( os.str() );
```
Comment (ihhub) [inline src/fheroes2/game/game_io.cpp:139]: Since we're at `Game` namespace we could remove `Game::`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @eos428 , I left few comments here. Could you please take a look?

File: src/fheroes2/game/game_campaign.cpp (hunk 1 above)
Comment (eos428) [reply src/fheroes2/game/game_campaign.cpp:35]: Probably no as of now, I forgot to remove this one when I moved the code to game_io

File: src/fheroes2/game/game_io.cpp (hunk 3 above)
Comment (eos428) [reply src/fheroes2/game/game_io.cpp:134]: I think in this case we just want to change something like `Good 1` to `Good_1`. In this case, would the `std::replace_if` below be enough?

**Comment by ihhub** [comment]
@eos428 , thank you very much for this improvement!

**Pull request merged**